kubernetes/provider.go

```diff
@@ -135,9 +135,11 @@ func Provider() terraform.ResourceProvider {
 		},
 
 		DataSourcesMap: map[string]*schema.Resource{
-			"kubernetes_secret":        dataSourceKubernetesSecret(),
-			"kubernetes_service":       dataSourceKubernetesService(),
-			"kubernetes_storage_class": dataSourceKubernetesStorageClass(),
+			"kubernetes_all_namespaces": dataSourceKubernetesAllNamespaces(),
```

Comment:
I did contemplate that possibility of future filters, so that's a valid point you're making.
Do you see filtering based on labels / annotations as something useful to have?
Can't think of anything else you'd want to filter on.

Reply:
Main use case I imagined was to filter on the `metadata.0.name` value, e.g. applying things like default network policies to namespaces matching `.*-public` or `.*-private`, but labels and annotations are good candidates too.